tests/functions.py

```diff
@@ -0,0 +1,72 @@
+from os import listdir, walk
+from os.path import isdir, exists, join, abspath, basename, dirname
+from random import random
+import datalad.api as api
+
+
+def recurse(directory, odds):
+
+    # Get all file names in directory
+    files = listdir(directory)
+
+    # Loop throw every file
+    for file_name in files:
+
+        # If the file name is .git or .datalad, ignore
+        if file_name == ".git" or file_name == ".datalad":
+            continue
+
+        full_path = join(directory, file_name)
+
+        # If the file is a directory
+        if isdir(full_path):
+
+            result = recurse(full_path, odds)
+
+            if result != "All good":
+                return result
```

Comment:
I think you could just `return recurse(full_path, odds)`, since you're returning "All good" later anyway.